km61/Servatmand_Mariam/laboratory1.sql

```diff
@@ -1,2 +1,129 @@
 -- LABORATORY WORK 1
 -- BY Servatmand_Mariam
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+робити вибірки з таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER Servatmand INDENTIFIED BY Servatmand
+DEFAULT TABLESPACE USERS
+TEMPORARY TABLESPACE TEMP
+
+ALTER Servatmand QUOTA 100 MB ON USERS
+ GRANT "CONNECT" TO Servatmand
+GRANT SELECT ANY TABLE TO USER
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Громадянин України має власне житло та автомобіль.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+GRANT INSERT TO Servatmand
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть постачальника, що продав найдешевший товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+SELECT vend_name
+where vend_id.vendors=vend_id.products and prod_price=min
+FROM VENDORS
+
+
+PROJECT VENDORS
+where vend_id.vendors=vend_id.products and prod_price=min
+{ vend_name}
+
```

Comment:
SELECT VENDORS.vend_name, ORDERITEMS.item_price
FROM VENDORS, PRODUCTS, ORDERITEMS
WHERE 
    VENDORS.vend_id = PRODUCTS.vend_id and
    PRODUCTS.prod_id = ORDERITEMS.prod_id and
    ORDERITEMS.item_price IN (SELECT MIN(ORDERITEMS.item_price) FROM ORDERITEMS  );